### PR TITLE
doc(notebook) : add more basic operations and some examples

### PR DESCRIPTION
**Pull request by Tyooughtul**
Add execution outputs to basic-tensor-op.ipynb
- Add outputs for trigonometric, reduction and comparison operations
- Add operation summary and PyTorch API comparison

Refactor plots.ipynb for autodiff and gradient descent
- Rewrite notebook to demonstrate autodiff and gradient flow
- Add GD from scratch and linear regression examples
- Update dependencies and remove unused plotting code

### Checklist

- [x] Confirmed that `cargo run-checks` command has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs
#2191 
_Provide links to relevant issues and dependent PRs._

### Changes

_Summarize the problem being addressed and your solution._

This commit has been a great learning experience for me with Burn. During the process, I used Grok to quickly locate files and find the interfaces I needed. I also had it generate some summaries, which I revised slightly, and I hope they will help future learners.

### Testing

_Describe how these changes have been tested._

### CODE REVIEW

File: examples/notebook/basic-tensor-op.ipynb [hunk 1]
```diff
@@ -40,83 +41,1138 @@
...
+    "- **Reduction Operations**: sum, mean, prod, max, min\n",
+    "- **Comparison**: greater, lower, equal, mask_where, mask_fill\n",
+    "\n",
+    "These are the fundamental building blocks for building neural networks in Burn!\n",
+    "\n",
+    "Note: Burn's API differs from PyTorch in several ways:\n",
+    "- Uses Rust's type system with explicit dimensions\n",
+    "- Method names like `lower` instead of `lt`, `powf_scalar` instead of `pow`\n",
+    "- Requires explicit device management and backend specification\n",
+    "- Distribution parameters use tuple syntax: `Normal(mean, std)` instead of keyword arguments"
```
Comment (laggui) [inline examples/notebook/basic-tensor-op.ipynb]: I think we can remove this part of the conclusion

File: examples/notebook/autodiff.ipynb
Comment (laggui) [inline examples/notebook/autodiff.ipynb:1]: I think we should rename this notebook to e.g. `autodiff.ipynb`

File: examples/notebook/plots.ipynb [hunk 2]
```diff
@@ -41,53 +36,551 @@
...
+    "    2.0 * x + 1.0 + noise.clone().slice([i..i+1]).into_scalar().elem::<f32>()\n",
+    "}).collect();\n",
+    "\n",
+    "let x: Tensor<B, 2> = TensorData::new(x_data.clone(), [num_samples, 1]).into();\n",
+    "let y: Tensor<B, 2> = TensorData::new(y_data.clone(), [num_samples, 1]).into();\n",
+    "\n",
+    "println!(\"Generated {} data points\", num_samples);\n",
+    "println!(\"True relationship: y = 2x + 1\");\n",
+    "println!(\"First 5 x values: {:?}\", &x_data[0..5]);\n",
+    "println!(\"First 5 y values: {:?}\", &y_data[0..5]);\n"
```
Comment (laggui) [inline examples/notebook/plots.ipynb]: Might as well use tensor ops to generate the target values!

```rust
// Generate synthetic data: y = 2x + 1 + noise
let num_samples = 100;
let x_data = TensorData::new((0..num_samples).map(|i| i as f32 / 10.0).collect(), [num_samples, 1]);
// Generate noise using Burn's random tensor
let noise = Tensor::<B, 2>::random([num_samples, 1], Distribution::Uniform(-0.25, 0.25), &device);

let x = Tensor::<B, 2>::from(x_data);
let y: Tensor<B, 2> = 2 * x.clone() + 1 + noise;

println!("Generated {} data points", num_samples);
println!("True relationship: y = 2x + 1");
println!("First 5 x values: {}", x.clone().slice(0..5));
println!("First 5 y values: {}", y.clone().slice(0..5));
```

**Comment by Tyooughtul** [comment]
@laggui @antimora Thanks for the review! I've resolved the request changes. Please take another look when you have time 😊
